include/directx/d3dx12.h

```diff
@@ -4037,6 +4037,665 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+            m_dOptions.TiledResourcesTier = D3D12_TILED_RESOURCES_TIER_NOT_SUPPORTED;
+            m_dOptions.ResourceBindingTier = (D3D12_RESOURCE_BINDING_TIER)0;
+            m_dOptions.PSSpecifiedStencilRefSupported = false;
+            m_dOptions.TypedUAVLoadAdditionalFormats = false;
+            m_dOptions.ROVsSupported = false;
+            m_dOptions.ConservativeRasterizationTier = D3D12_CONSERVATIVE_RASTERIZATION_TIER_NOT_SUPPORTED;
+            m_dOptions.MaxGPUVirtualAddressBitsPerResource = 0;
+            m_dOptions.StandardSwizzle64KBSupported = false;
+            m_dOptions.CrossNodeSharingTier = D3D12_CROSS_NODE_SHARING_TIER_NOT_SUPPORTED;
+            m_dOptions.CrossAdapterRowMajorTextureSupported = false;
+            m_dOptions.VPAndRTArrayIndexFromAnyShaderFeedingRasterizerSupportedWithoutGSEmulation = false;
+            m_dOptions.ResourceHeapTier = (D3D12_RESOURCE_HEAP_TIER)0;
+        }
+
+        if (INITIALIZE_FAILED(D3D12_FEATURE_GPU_VIRTUAL_ADDRESS_SUPPORT, m_dGPUVASupport)) {
+            m_dGPUVASupport.MaxGPUVirtualAddressBitsPerProcess = 0;
+            m_dGPUVASupport.MaxGPUVirtualAddressBitsPerResource = 0;
+        }
+
+        if (INITIALIZE_FAILED(D3D12_FEATURE_D3D12_OPTIONS1, m_dOptions1)) {
+            m_dOptions1.WaveOps = false;
+            m_dOptions1.WaveLaneCountMax = 0;
+            m_dOptions1.WaveLaneCountMin = 0;
+            m_dOptions1.TotalLaneCount = 0;
+            m_dOptions1.ExpandedComputeResourceStates = 0;
+            m_dOptions1.Int64ShaderOps = 0;
+        }
+
+        if (INITIALIZE_FAILED(D3D12_FEATURE_D3D12_OPTIONS2, m_dOptions2)) {
+            m_dOptions2.DepthBoundsTestSupported = false;
+            m_dOptions2.ProgrammableSamplePositionsTier = D3D12_PROGRAMMABLE_SAMPLE_POSITIONS_TIER_NOT_SUPPORTED;
+        }
+
+        if (INITIALIZE_FAILED(D3D12_FEATURE_SHADER_CACHE, m_dShaderCache)) {
+            m_dShaderCache.SupportFlags = D3D12_SHADER_CACHE_SUPPORT_NONE;
+        }
+
+        if (INITIALIZE_FAILED(D3D12_FEATURE_D3D12_OPTIONS3, m_dOptions3)) {
+            m_dOptions3.CopyQueueTimestampQueriesSupported = false;
+            m_dOptions3.CastingFullyTypedFormatSupported = false;
+            m_dOptions3.WriteBufferImmediateSupportFlags = D3D12_COMMAND_LIST_SUPPORT_FLAG_NONE;
+            m_dOptions3.ViewInstancingTier = D3D12_VIEW_INSTANCING_TIER_NOT_SUPPORTED;
+            m_dOptions3.BarycentricsSupported = false;
+        }
+
+        if (INITIALIZE_FAILED(D3D12_FEATURE_EXISTING_HEAPS, m_dExistingHeaps)) {
+            m_dExistingHeaps.Supported = false;
+        }
+
+        if (INITIALIZE_FAILED(D3D12_FEATURE_D3D12_OPTIONS4, m_dOptions4)) {
+            m_dOptions4.MSAA64KBAlignedTextureSupported = false;
+            m_dOptions4.Native16BitShaderOpsSupported = false;
+            m_dOptions4.SharedResourceCompatibilityTier = (D3D12_SHARED_RESOURCE_COMPATIBILITY_TIER)0; // TODO: 0 is occupied. See if needs another value

```

Comment:
0 is sometimes a valid entry for some feature support data, particularly *_TIER enums that have a TIER_0 entry. Is this the same as NOT_SUPPORTED? If not, what default value would you suggest if the initialization fails?

Reply:
The I guess it is safe to use TIER_0 as a fallback value when the initialization fails, as long as we're running with a valid D3D12 implementation?

Reply:
I'm using TIER_0 as default values for these caps. They'll be up in the next commit. Let me know if this can be an issue.

Reply:
Resolved.

Reply:
Yeah, sounds good.